app/models/company.rb

```diff
@@ -2,4 +2,33 @@ class Company < ActiveRecord::Base
   validates :name, :city, presence: true
   validates :name, uniqueness: true
   has_many :jobs
+  has_many :contacts
+
+  def self.sort_by_location
+    order(:city)
+  end
+
+  def self.top_three_companies_by_interest
```

Comment:
these two methods (this and `list_number_of_jobs_by_location` have a lot of Ruby and a lot of logic in each. can we break the content down in each even further? also - can we leverage some AR methods like group, joins, count, etc.?